### PR TITLE
Add new rule QuoteMapKeysRule, with tests

### LINKED ISSUE #12: Add a lint rule for unquoted strings in map keys
It's generally a bad practice to use unquoted strings as map keys. Some tokens (like `green`) parse as colors rather than strings, which means that if the keys are used in selector names, they won't necessarily emit as text (for example, `green` is emitted as `#0f0` in compressed mode). It would be good to have a lint that encourages users to quote their map keys.

### PR DESCRIPTION
**Pull request by srawlins**
Fixes #12 

I also updated CHANGELOG to use the under_scored_string names for each lint rule, as that will probably be how users see the rules (output messages, and enabling/disabling in a config file).

### CODE REVIEW

File: lib/src/rules/quote_map_keys.dart [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import 'package:sass/src/ast/sass.dart';
+
+import '../lint.dart';
+import '../rule.dart';
+
+/// A lint rule that reports on unquoted map keys in a map literal.
+///
+/// Some tokens (like `green`) parse as colors rather than strings. It is safer
+/// to always quote map keys.
+/// to browsers, in the CSS output.
```
Comment (nex3) [inline lib/src/rules/quote_map_keys.dart]: Nit: Extra line

File: test/rule_test.dart [hunk 2]
```diff
@@ -271,6 +271,25 @@ void main() {
     expect(lint.column, 16);
   });
 
+  test('reports lint when boolean is found in a map literal', () {
```
Comment (nex3) [inline test/rule_test.dart]: How is this test passing right now? It doesn't seem like this lint would get triggered by the rule you wrote.

File: lib/src/rules/quote_map_keys.dart (hunk 1 above)
Comment (srawlins) [reply lib/src/rules/quote_map_keys.dart]: Done.

File: test/rule_test.dart (hunk 2 above)
Comment (srawlins) [reply test/rule_test.dart]: This is a test for the `visitMapExpression` implementation in `lib/src/rule.dart`.

**Pull request merged**